MushroomObserver/mushroom-observer
# Split out classes from app/classes/auto_complete.rb

so it plays better with the autoloader.

To test:
Load http://localhost:3000/ajax/auto_complete/name/b
Modify some of the rails code
Reload http://localhost:3000/ajax/auto_complete/name/b

On main the above should fail.  On the branch it should succeed.  You can replicate the same behavior by trying to use the autoloader from the Create Observation page instead of loading the AJAX URLs.
